proxy/enfuse/socket_test.go

```diff
@@ -0,0 +1,83 @@
+package enfuse_test
+
+import (
+	"fmt"
+	"github.com/enfabrica/enkit/lib/knetwork"
+	"github.com/enfabrica/enkit/proxy/enfuse"
+	"github.com/enfabrica/enkit/proxy/enfuse/testserver"
+	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"strings"
+	"testing"
+)
+
+type testClientConn struct {
+	Name string
+	Conn *websocket.Conn
+	Shim *enfuse.SocketShim
+}
+
+var testConnections []*testClientConn = []*testClientConn{
+	{
+		Name: "pluto",
+	},
+	{
+		Name: "donald",
+	},
+	{
+		Name: "mickey",
+	},
+	{
+		Name: "minnie mouse",
+	},
+	{
+		Name: "goofy",
+	},
+}
+
+func TestSocketShim(t *testing.T) {
+	s := testserver.NewWebSocketBasicClientServer(t)
+	defer s.Close()
+
+	// Dial all websocket clients to the httptest.Server
+	url := strings.Replace(s.URL, "http", "ws", -1)
```

Comment:
Just the first http needs to be replaced, right?

Reply:
I suppose putting the -1 is arbitrary, replace All looks prettier

Reply:
...but only the first should be replaced, right? This rewrites `http://somehttpserver.com` into `ws://somewsserver.com` which might be OK in the context of the test but definitely dials a different server

Reply:
This rewrites the httptest.Server, which is guaranteed to be an ip and random port. I don't think it'll ever change to be a url.